reputation-commitments/src/tests.rs

```diff
@@ -0,0 +1,399 @@
+				&bob.clone(),
+				cid,
+				11,
+				0,
+				None
+			));
+			assert_eq!(EncointerReputationCommitments::commitments((cid, 11), (0, &bob)), None);
+			assert_eq!(
+				last_event::<TestRuntime>(),
+				Some(Event::CommitedReputation(cid, 11, 0, bob.clone(), None).into())
+			);
+
+			// same reputation, different purpose
+			// use extrinsic
+			assert_ok!(EncointerReputationCommitments::commit_reputation(
+				RuntimeOrigin::signed(alice.clone()),
+				cid,
+				11,
+				1,
+				None
+			));
+			assert_eq!(EncointerReputationCommitments::commitments((cid, 11), (1, &alice)), None);
+			assert_eq!(
+				last_event::<TestRuntime>(),
+				Some(Event::CommitedReputation(cid, 11, 1, alice.clone(), None).into())
+			);
+
+			// commit with hash
+			// use extrinsic
+			let hash = "Some value".using_encoded(BlakeTwo256::hash);
+
+			assert_ok!(EncointerReputationCommitments::commit_reputation(
+				RuntimeOrigin::signed(alice.clone()),
+				cid2,
+				12,
+				0,
+				Some(hash)
+			));
+			assert_eq!(
+				EncointerReputationCommitments::commitments((cid2, 12), (0, &alice)),
+				Some(hash)
+			);
+			assert_eq!(
+				last_event::<TestRuntime>(),
+				Some(Event::CommitedReputation(cid2, 12, 0, alice.clone(), Some(hash)).into())
+			);
+
+			// already commited
+			assert_err!(
+				EncointerReputationCommitments::do_commit_reputation(
+					&alice.clone(),
+					cid,
+					11,
+					0,
+					None
+				),
+				Error::<TestRuntime>::AlreadyCommited
+			);
+
+			// no reputation
+			assert_err!(
+				EncointerReputationCommitments::do_commit_reputation(
+					&alice.clone(),
+					cid,
+					13,
+					0,
+					None
+				),
+				Error::<TestRuntime>::NoReputation
+			);
+
+			assert_err!(
+				EncointerReputationCommitments::do_commit_reputation(
+					&alice.clone(),
+					cid2,
+					11,
+					0,
+					None
+				),
+				Error::<TestRuntime>::NoReputation
+			);
+
+			// inexistent purpose
+			assert_err!(
+				EncointerReputationCommitments::do_commit_reputation(
+					&alice.clone(),
+					cid,
+					11,
+					2,
+					None
+				),
+				Error::<TestRuntime>::InexistentPurpose
+			);
+		})
+	});
+}
+
+#[test]
+fn purging_works() {
+	new_test_ext().execute_with(|| {
+		let mut ext = new_test_ext();
+		let alice = AccountId::from(AccountKeyring::Alice);
+		let bob = AccountId::from(AccountKeyring::Bob);
+		let cid = register_test_community::<TestRuntime>(None, 0.0, 0.0);
+		let cid2 = register_test_community::<TestRuntime>(None, 10.0, 10.0);
+
+		//ext.insert(community_identifiers(), vec![cid, cid2].encode());
+
+		ext.insert(participant_reputation((cid, 1), &alice), Reputation::VerifiedUnlinked.encode());
+		ext.insert(
+			participant_reputation((cid2, 1), &alice),
+			Reputation::VerifiedUnlinked.encode(),
+		);
+		ext.insert(participant_reputation((cid, 1), &bob), Reputation::VerifiedUnlinked.encode());
+		ext.insert(participant_reputation((cid2, 1), &bob), Reputation::VerifiedUnlinked.encode());
```

Comment:
cheeky Bob managed to be in two communities for the same cycle. Not relevant, but strange....